app/controllers/articles_controller.rb

```diff
@@ -184,6 +184,18 @@ def admin_unpublish
     end
   end
 
+  def admin_featured_toggle
+    authorize @article
+
+    @article.featured = params[:article][:featured].to_i == 1
```

Comment:
A `nil == "1"` check works and doesn't rely on the `nil.to_i == 0` behavior. So maybe this instead?

```suggestion
    @article.featured = params.dig(:article, :featured) == "1"
```